src/compiler/crystal/compiler.cr

```diff
@@ -86,9 +86,9 @@ module Crystal
     # A `ProgressTracker` object which tracks compilation progress.
     property progress_tracker = ProgressTracker.new
 
-    # Target triple to use in the compilation.
+    # Codegen target to use in the compilation.
     # If not set, asks LLVM the default one for the current machine.
-    property target_triple : String?
+    property codegen_target = Codegen::Target.new
```

Comment:
Could this be made a lazy property?

Reply:
Because the comments say it will be the default one unless you set one. From that I expect it to be lazily created (unless you set one). But it's fine if that was not the intention.